File: src/week2/readability/readability.c
#include "../../../lib/cs50.h"
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
// #include <stdbool.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // get the target content
    string content = get_string("Text: ");
    float index = 5.88 * ((float)count_letters(content) / (float)count_words(content)) - 29.6 * ((float)count_sentences(content) / (float)count_words(content)) - 15.8;
    // printf("%d,%d,%d\n", count_letters(content), count_words(content), count_sentences(content));
    int ans = round(index);
    if (ans < 1)
    {
        printf("Before Grade 1");
    }
    else if (ans >= 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %d", ans);
    }
}

int count_letters(string text)
{
    int letters_count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]) != 0)
        {
            letters_count++;
        }
    }
    return letters_count;
}

int count_words(string text)
{
    // if had cross over a space from alphabeta
    bool has_separated = false;
    int words_count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words_count++;
        }
    }
    words_count++;
    return words_count;
}

int count_sentences(string text)
{
    int sentences_count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences_count++;
        }
    }
    return sentences_count;
}

File: src/week6/sentimental-readability/readability.py
# TODO

def main():
    content = input("Text: ")
    letters_count = count_letters(content)
    words_count = count_words(content)
    sentences_count = count_sentences(content)
    L = 100 * letters_count / words_count
    S = 100 * sentences_count / words_count
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    letters_count = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters_count += 1
    return letters_count


def count_words(text):
    words_count = len(text.split())
    return words_count


def count_sentences(text):
    sentences_count = 0
    for i in range(len(text)):
        if text[i] in ['.', '!', '?']:
            sentences_count += 1
    return sentences_count


if __name__ == "__main__":
    main()
